docs/source/entities-advanced.mdx

```diff
@@ -0,0 +1,535 @@
+    }
+  }
+}
+```
+
+### Using `@requires` with object subfields
+
+If a computed field `@requires` a field that returns an object type, you also specify which _subfields_ of that object are required. You list those subfields with the following syntax:
+
+```graphql{4}
+# Shipping subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  dimensions: ProductDimensions @external
+  shippingEstimate: String @requires(fields: "dimensions { size weight }")
+}
+```
+
+In this modification of the previous example, `size` and `weight` are now subfields of a `ProductDimensions` object. Note that the `ProductDimensions` type must be defined in both the Products _and_ Shipping subgraphs for this to be valid.
+
+## Resolving another subgraph's field
+
+By default, exactly one subgraph is responsible for resolving each field in your supergraph schema (with some exceptions, like `@key` fields). But sometimes, _multiple_ subgraphs are able to resolve a particular entity field, because all of those subgraphs have access to a particular data store. For example, an Inventory subgraph and a Products subgraph might both have access to the database that stores all product-related data.
+
+You can enable multiple subgraphs to resolve a particular entity field. **This is a completely optional optimization.** When the gateway plans a query's execution, it looks at which fields are available from each subgraph. It can then attempt to optimize performance by executing the query across the fewest subgraphs needed to access all required fields.
+
+You achieve this with one of the following directives:
+
+* [`@shareable`](#using-shareable)
+* [`@provides`](#using-provides)
+
+Which directive you use depends on the following logic:
+
+```mermaid
+graph TB;
+  resolve("Can my subgraph always resolve this field,<br/>or only from certain query paths?");
+  resolve--Always-->shareable("@shareable");
+  resolve--"Only certain query paths"-->provides("@provides");
+```
+
+If you aren't sure whether your subgraph can always resolve a field, see [Using `@provides`](#using-provides) for an example of a subgraph that _can't_.
+
+> ⚠️ **Important:** If multiple subgraphs can resolve a field, **make sure each subgraph's resolver for that field behaves identically.** Otherwise, queries might return inconsistent results depending on which subgraph resolves the field.
+
+### Using `@shareable`
+
+The `@shareable` directive indicates that a particular field can be resolved by more than one subgraph:
+
+<CodeColumns>
+
+```graphql{4}
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @shareable
+  price: Int
+}
+```
+
+```graphql{4}
+# Inventory subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @shareable
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+In this example, both the Products and Inventory subgraphs can resolve `Product.name`. This increases the likelihood that a client's query can be resolved by fewer total subgraphs.
```

Comment:
I tripped a bit on this sentence. I'd probably rephrase saying something along the lines of "This is usually done as an optimisation: as multiple subgraphs are able to resolve the field, queries involving that field may be resolved by querying fewer total subgraphs". 

But I would also warn about the flip-side of the tradeoff involved. Namely, I'd first insist once against that all subgraphs resolving the field must resolve it the same way. It's a pretty important point imo and I'd suggest expanding on it; maybe with a full subsection, giving an example of what to do and what _not_ to do to clarify what we mean by "resolve the same way"). And then essentially say that not only those resolvers must work the same way initially, but they should be kept in sync if evolved, and that this require care and attention, and so that is why we advise against premature optimisation (but feel free to avoid the term "premature optimisation").

Lastly, as mentioned previously, `@shareable` is not only for entities: but all that I mention in that section does _only_ really apply to `@shareable` when applies to fields of entities. When `@shareable` is used with value types, it's not really an optimisation, it's just the only way in fed 2 to enable a shared value type and that's pretty common.

Reply:
👍 Created a dedicated section describing these concerns and link to it from the top of each section about `@shareable` and `@provides`. Can hopefully expand it further prior to release